monitoring/stats_dump_scheduler.h

```diff
@@ -0,0 +1,50 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#pragma once
+
+#ifndef ROCKSDB_LITE
+
+#include "rocksdb/db.h"
+#include "util/timer.h"
+
+namespace ROCKSDB_NAMESPACE {
+
+class StatsDumpScheduler {
+ public:
+  static std::shared_ptr<StatsDumpScheduler> Default();
+
+  ~StatsDumpScheduler();
+
+  void Register(DB* db, unsigned int stats_dump_period_sec,
```

Comment:
Is `StatsDumpScheduler` going to be public API? If not, I think `DBImpl* db` suffices.